src/main/scala/net/snowflake/spark/snowflake/SnowflakeJDBCWrapper.scala

```diff
@@ -84,7 +84,14 @@ private[snowflake] class JDBCWrapper {
         val isSigned = rsmd.isSigned(i + 1)
         val nullable = rsmd.isNullable(i + 1) != ResultSetMetaData.columnNoNulls
         val columnType = getCatalystType(dataType, fieldSize, fieldScale, isSigned)
-        fields(i) = StructField(columnName, columnType, nullable)
+        fields(i) =
+          StructField(
+            if(columnName == columnName.toUpperCase) // Add quotes around column names if not all uppercase
```

Comment:
Don't we need something for names with special characters (space, quote, doublequote etc) ?

We should test columns like that.